packages/theme/src/theme.ts

```diff
@@ -211,17 +254,18 @@ export class Theme extends HTMLElement {
         kind: FragmentType,
         styles: CSSResult
     ): void {
-        this.themeFragments.set(name, { kind, styles });
-    }
-
-    static themeFragment(
-        name: FragmentName
-    ): { kind: FragmentType; styles: CSSResult } {
-        const fragment = this.themeFragments.get(name);
-        if (!fragment) {
-            throw new Error(`Unknown theme fragment '${name}'`);
+        if (!this.themeFragmentsByKind.has(kind)) {
+            this.themeFragmentsByKind.set(kind, new Map());
+            (this.themeFragmentsByKind.get(kind) as FragmentMap).set(
```

Comment:
Not sure you need these casts anymore?

Reply:
The default type of `this.themeFragmentsByKind.get(kind)` is `FragmentMap | null`, so as best as I understand TS I either need this or to set a variable and test its presence (both here and below), but that doesn't compile away, right? Is there a nicer in between version?

Reply:
Ah I see. You could use the non-null operator, or you could just write this a little, something like this seems easier to read and maintain, not sure how it stacks up in terms of compiled byte size though:

```typescript

        let fragmentMap = this.themeFragmentsByKind.get(kind);
        if (!fragmentMap) {
            fragmentMap = new Map();
            this.themeFragmentsByKind.set(kind, new Map());
            // we're adding our first fragment for this kind set as default
            fragmentMap.set('default', { name, styles });
            Theme.defaultFragments.add(name);
        }
        fragmentMap.set(name, {
            name,
            styles,
        });
```